src/common/re2_regex.cpp

```
@@ -7,56 +7,90 @@
 }
 
 bool RegexMatch(const std::string &input, Match &match, const Regex &regex) {
-	return RegexSearchInternal(input.c_str(), match, regex, RE2::ANCHOR_BOTH, 0, input.size());
+	auto input_sz = input.size();
+	return RegexSearchInternal(input.c_str(), input_sz, match, regex.GetRegex(), RE2::ANCHOR_BOTH, 0, input_sz);
 }
 
 bool RegexMatch(const char *start, const char *end, Match &match, const Regex &regex) {
-	return RegexSearchInternal(start, match, regex, RE2::ANCHOR_BOTH, 0,
-	                           duckdb::UnsafeNumericCast<size_t>(end - start));
+	auto sz = duckdb::UnsafeNumericCast<size_t>(end - start);
+	return RegexSearchInternal(start, sz, match, regex.GetRegex(), RE2::ANCHOR_BOTH, 0, sz);
 }
 
 bool RegexMatch(const std::string &input, const Regex &regex) {
 	Match nop_match;
-	return RegexSearchInternal(input.c_str(), nop_match, regex, RE2::ANCHOR_BOTH, 0, input.size());
+	auto input_sz = input.size();
+	return RegexSearchInternal(input.c_str(), input_sz, nop_match, regex.GetRegex(), RE2::ANCHOR_BOTH, 0, input_sz);
 }
 
 duckdb::vector<Match> RegexFindAll(const std::string &input, const Regex &regex) {
+	return RegexFindAll(input.c_str(), input.size(), regex.GetRegex());
+}
+
+duckdb::vector<Match> RegexFindAll(const char *input_data, size_t input_size, const RE2 &regex) {
 	duckdb::vector<Match> matches;
 	size_t position = 0;
 	Match match;
-	while (RegexSearchInternal(input.c_str(), match, regex, RE2::UNANCHORED, position, input.size())) {
-		position += match.position(0) + match.length(0);
+	while (RegexSearchInternal(input_data, input_size, match, regex, RE2::UNANCHORED, position, input_size)) {
+		if (match.length(0)) {
+			position = match.position(0) + match.length(0);
+		} else { // match.length(0) == 0
+			auto next_char_length = GetMultibyteCharLength(input_data[match.position(0)]);
+			if (!next_char_length) {
+				// invalid UTF-8 leading byte
```

should we throw an error on invalid utf-8?